### ISSUE TITLE
Entity deletion does not work(occasionally)

### ISSUE THREAD

**Issue by Alexc0007** [sensu/sensu-go#3040]
Hi , 
im using sensu-go-backend 5.8 
and using API to delete entities when they are no longer relevant to my production env.

sometimes it happens that when i try to delete an entity - i receive 204 for the delete request but the entity is not deleted.

since my environment is very dynamic - it happens that servers go up and down a lot and many deletions are required.

i use the api as described in the documentation: 
curl -X DELETE \
-H "Authorization: Bearer $SENSU_TOKEN" \
http://127.0.0.1:8080/api/core/v2/namespaces/default/entities/server1

## Expected Behavior
when i use api to delete entity - i expect it to be removed from sensu.

## Current Behavior
99% of the time everything works as expected , but it sometimes happens that API return 204 but not deleting the entity.

## Possible Solution
im not sure why it happens , since API return 204.
i applied debug mode for the logs and according to what is documented on the logs - it process the deletion successfully.
but still - doesnt delete the entity.

i manage this situation by creating a retry mechanism that tries to delete an entity , then use api to check if entity is present , if it is , i try to delete again.
i use maximum 3 retries and now i get 100% entity deletion when its required.

so from sensu side the solution should probably be that the API return 204 only after a verification that the entity was deleted.

## Steps to Reproduce (for bugs)
i cant point at a specific situation when it happens , it just happens randomly.

## Your Environment
my sensu backend version: sensu-go-backend-5.8.0-2735.x86_64
my sensu agent version: sensu-go-agent-5.6.0-2141.x86_64
linux version: 4.14.121-109.96.amzn2.x86_64 (Amazon Linux 2)
sensu is installed on the host , not docker.

**Comment by echlebek** [comment]
It's unlikely that delete only works some of the time.

Since entities are recreated by agents, there could be a race condition in your system, where agents are not shutting down until after you issue the delete, resulting in the appearance that the delete is not working.

**Comment by Alexc0007** [comment]
its possible you are correct...
either way , after applying a retry and check mechanism everything seem to work fine for me.
ill close this issue
